src/compiler/marks.ts

```diff
@@ -357,8 +357,53 @@ export namespace bar {
   }
 
   export function labels(model: Model) {
-    // TODO(#64): fill this method
-    return undefined;
+    // TODO(#64):
+    // √ horizontal support
+    // - vertical support
+    // - color support
+    // - container height/stretching consideration
+    // - consider defaults for offset
+
+    var l: any = {};
+    const orient = model.marksConfig('orient');
+
+    l.align = { value: (orient !== 'horizontal') ? "center" : "left" };
+
+    l.text = { field: model.has(LABEL) ? model.field(LABEL) : ((orient !== 'horizontal') ? model.field(Y) : model.field(X)) };
```

Comment:
ah you're right. I should've looked more closely. I wonder if there's a case for wanting to apply a label without being explicit with field. In the case of a bar chart with a quantitative variable and a ordinal/categorical variable, I can't imagine the label would be of anything but the quantitative var. Either way, totally makes sense, I'll remove that conditional.